torch/cuda/random.py

```diff
@@ -101,15 +123,33 @@ def seed():
         If you are working with a multi-GPU model, this function will only initialize
         the seed on one GPU.  To initialize all GPUs, use :func:`seed_all`.
     """
-    _lazy_call(lambda: _C._cuda_seed())
+    def cb():
+        idx = current_device()
+        default_generator = default_cuda_generators[idx]
+        default_generator.seed()
+
+    _lazy_call(cb)
 
 
 def seed_all():
     r"""Sets the seed for generating random numbers to a random number on all GPUs.
     It's safe to call this function if CUDA is not available; in that
     case, it is silently ignored.
     """
-    _lazy_call(lambda: _C._cuda_seedAll())
+    def cb():
+        random_seed = 0
+        seeded = False
+        for i in range(device_count()):
+            with device_ctx_manager(i):
+                default_generator = default_cuda_generators[i]
+                if not seeded:
+                    default_generator.seed()
+                    random_seed = default_generator.initial_seed()
+                    seeded = True
```

Comment:
Err... do you really want the RNGs on each GPU to be lockstep? That sounds surprising to me.

Reply:
I take it back, this is apparently what we do today lol.